host/libraries/libbladeRF/src/board/bladerf1/bladerf1.c

```diff
@@ -2066,8 +2066,19 @@ static int bladerf1_set_frequency(struct bladerf *dev,
             break;
 
         case BLADERF_TUNING_MODE_FPGA: {
-            status = dev->board->schedule_retune(dev, ch, BLADERF_RETUNE_NOW,
-                                                 frequency, NULL);
+            if (attached == BLADERF_XB_200) {
+                bladerf_quick_tune qt;
+                status = bladerf_get_quick_tune(dev, ch, &qt);
+                if (status != 0) {
+                    return status;
+                }
+                
+                status = dev->board->schedule_retune(dev, ch, BLADERF_RETUNE_NOW, 0, &qt);
+            }
```

Comment:
Any chance this right curly bracket } can be on the same line as the else statement?

Reply:
I'm trying to make sense of this.  If the XB200 is attached, how does bladerf_get_quick_tune() return a bladerf_quick_tune that is for the correct new frequency?

Reply:
I meant to fix this issue: https://github.com/Nuand/bladeRF/issues/882
But i do not have equipment here  to retest at the moment. Maybe i do not understand the error correctly.